src/mlpack/methods/preprocess/preprocess_string_util.cpp

```diff
@@ -0,0 +1,159 @@
+/**
+ * @file preprocess_string_util.cpp
+ * @author Jeffin Sam
+ *
+ * A CLI executable to encode string dataset.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#include <mlpack/prereqs.hpp>
+#include <mlpack/core/util/mlpack_main.hpp>
+#include <mlpack/core/math/random.hpp>
+#include <mlpack/core/util/cli.hpp>
+#include <mlpack/core/data/extension.hpp>
+#include <unordered_set>
+#include <mlpack/core/data/string_encoding.hpp>
+#include <mlpack/core/data/tokenizers/split_by_any_of.hpp>
+#include <mlpack/core/data/string_encoding_policies/dictionary_encoding_policy.hpp>
+#include <mlpack/core/data/string_encoding_policies/bag_of_words_encoding_policy.hpp>
+#include <mlpack/core/data/string_encoding_policies/tf_idf_encoding_policy.hpp>
+
+using namespace mlpack;
+using namespace mlpack::util;
+using namespace arma;
+using namespace std;
+
+/**
+ * Function neccessary to create a vector<vector<string>> by readin
+ * the contents of a file.
+ *
+ * @param filename Name of the file whose contents need to be preproccessed.
+ * @param columnDelimiter Delimiter used to split the columns of file.
+ */
+static vector<vector<string>> CreateDataset(const string& filename,
```

Comment:
These functions are used in other files. They shouldn't be static. And I think it's a good idea to move them to a separate namespace in order to avoid conflicts.